### ISSUE TITLE
Strange order dependency between zoomTransformProperty and zoomScaleProperty

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics#295]
```
assert.js:25 Uncaught Error: Assertion failed: space for units label is negative or zero
    at window.assertions.assertFunction (assert.js:25:13)
    at new RulerNode (RulerNode.js? [sm]:147:21)
    at createRulerNode (GORulerNode.ts:340:10)
    at GORulerNode.ts:136:22
    at TinyProperty.emit (TinyEmitter.ts:93:9)
    at DerivedProperty._notifyListeners (Property.ts:289:23)
    at DerivedProperty.set (Property.ts:223:14)
    at DerivedProperty.getDerivedPropertyListener (DerivedProperty.ts:131:13)
    at TinyProperty.emit (TinyEmitter.ts:93:9)
    at NumberProperty._notifyListeners (Property.ts:289:23)
```

To reproduce manually:

1. Run the sim with `?ea`
2. Go to Lens screen
3. Pull a ruler out of toolbox
4. Press zoom-out (-) button
5. Press zoom-in (+) button. The assertion will fail.

**Comment by pixelzoom** [comment]
Scaling of rulers got messed up sometime since 1.1.0-dev.14 was published on 1/12/2022.  Their view length should not change when zooming in/out, only the scale on the ruler should change.  But the view length of the ruler is now changing.

For example, here's the horizontal ruler initially:

<img width="862" alt="screenshot_1527" src="https://user-images.githubusercontent.com/3046552/149986547-a53765a0-af01-466a-a04b-d49222d7d219.png">

... and after pressing the zoom-out button once:

<img width="870" alt="screenshot_1528" src="https://user-images.githubusercontent.com/3046552/149986584-a319532b-b5db-4637-8e48-6d98b823a779.png">

**Comment by pixelzoom** [comment]
Bisecting, I isolated this bug to https://github.com/phetsims/geometric-optics/commit/9802b4f10968dcb2a6146105951a2345aee107b8.

**Comment by pixelzoom** [comment]
As noted in the above commit:

```
    //TODO There's a strange order dependency between zoomTransformProperty and zoomScaleProperty. If
    // zoomScaleProperty is defined first, then zoomTransformProperty is incorrect, and rulers will be
    // messed up as in https://github.com/phetsims/geometric-optics/issues/295. This has something to
    // do with the fact both Properties call getAbsoluteZoomScale.
```

So there's more work to do here.  And this is another good reason to reimplement model-view transforms in https://github.com/phetsims/geometric-optics/issues/277.

**Comment by pixelzoom** [comment]
Addressed in #277, closing.
